lib/services/livesync/playground/preview-app-plugins-service.ts

```diff
@@ -9,10 +9,23 @@ import { PLATFORMS_DIR_NAME, PACKAGE_JSON_FILE_NAME } from "../../../constants";
 export class PreviewAppPluginsService implements IPreviewAppPluginsService {
 	private previewAppVersionWarnings: IDictionary<string[]> = {};
 
-	constructor(private $fs: IFileSystem,
+	constructor(private $errors: IErrors,
+		private $fs: IFileSystem,
 		private $logger: ILogger,
 		private $pluginsService: IPluginsService) { }
 
+	public getDeviceWarnings(device: Device): string[] {
```

Comment:
I don't expect getDeviceWarnings in PreviewAppPluginsService. It doesn't look like a plugins related logic.